src/libraries/System.Formats.Tar/src/System/Formats/Tar/GnuTarEntry.cs

```diff
@@ -33,6 +33,21 @@ public GnuTarEntry(TarEntryType entryType, string entryName)
         {
         }
 
+        /// <summary>
+        /// Initializes a new <see cref="GnuTarEntry"/> instance by converting the specified <paramref name="other"/> entry into the GNU format.
+        /// </summary>
+        public GnuTarEntry(TarEntry other)
+            : this(other._header, other._readerOfOrigin!)
+        {
+            if (_header._typeFlag == TarEntryType.V7RegularFile)
+            {
+                _header._typeFlag = TarEntryType.RegularFile;
+            }
+            TarHelpers.VerifyEntryTypeIsSupported(_header._typeFlag, TarEntryFormat.Gnu, forWriting: false);
```

Comment:
How do we know that this isn't being used for writing?

Reply:
This parameter has an ambiguous name. What I mean is "am I calling this method from inside a WriteEntry method?". It should be called `invokingInsideWriteEntry` or something like that. The reason is that with the code as it is, a `TarWriter` is created with a format, and only entries of such format are allowed. If you pass an entry of another format, it will be written in that format.

But we changed that behavior, and I will fix that in my _next_ PR, which will remove the `forWriting` parameter altogether, since now we can write entries in any format, and they should get saved in that format.